spec/importers/data_transformers/csv_transformer_spec.rb

```diff
@@ -5,28 +5,18 @@
   describe '#transform' do
     context 'with good data' do
       context 'headers in csv' do
-        let!(:file) { File.open("#{Rails.root}/spec/fixtures/fake_x2_assessments.csv") }
-        let(:headers_symbols) {[:local_id,:school_local_id,:assessment_date,:assessment_scale_score,:assessment_performance_level,:assessment_growth,:assessment_name,:assessment_subject,:assessment_test]}
+        # Return a File.read string just like the CsvDownloader class does:
+        let!(:file) { File.read("#{Rails.root}/spec/fixtures/fake_behavior_export.txt") }
+
         let(:transformer) { CsvTransformer.new }
         let(:output) { transformer.transform(file) }
         it 'returns a CSV' do
           expect(output).to be_a_kind_of CSV::Table
         end
         it 'has the correct headers' do
-          expect(output.headers).to match_array(headers_symbols)
-        end
-      end
-      context 'headers not in csv' do
-        let!(:file) { File.open("#{Rails.root}/spec/fixtures/fake_no_headers.csv") }
-        let(:headers) {["section_number","student_local_id","school_local_id","course_number","term_local_id","grade"]}
-        let(:headers_symbols) {[:section_number,:student_local_id,:school_local_id,:course_number,:term_local_id,:grade]}
-        let(:transformer) { CsvTransformer.new(headers:headers) }
-        let(:output) { transformer.transform(file) }
-        it 'returns a CSV' do
-          expect(output).to be_a_kind_of CSV::Table
-        end
-        it 'has the correct headers' do
-          expect(output.headers).to match_array(headers_symbols)
```

Comment:
Can you explain more about why is this removed?

Reply:
Ooooh, this is a great question! I assumed that the `headers:` option for CsvTransformer wasn't being used anywhere, but searching through the codebase shows me that was wrong because StudentSectionGradesImporter uses it:

```
#student_section_grades_importer.rb

def data_transformer
  CsvTransformer.new(headers:%w[section_number student_local_id school_local_id course_number term_local_id grade])
end
```

I'm curious why StudentSectionGradesImporter needs to do this, I'll dig into the export file / commit history to find out. In the meantime, you're write, this PR shouldn't remove the spec for this code path so I'll restore.

Reply:
The StudentSectionGradesImporter imports data from 'student_averages_export.txt', so I assumed we'd have a 'student_averages_export.sql' file under the /x2_exports directory, but I don't see any. I want to find that SQL file and put it under source control so we can see why the importer needs to use the `headers:` option . . . but that's a separate issue.